src/blockchain/electrum.rs

```diff
@@ -385,4 +386,116 @@ mod test {
 
         assert_eq!(wallet.get_balance().unwrap(), 50_000);
     }
+
+    #[test]
+    fn test_electrum_blockchain_factory_sync_with_stop_gaps() {
+        // Test whether Electrum blockchain syncs with expected behaviour given different `stop_gap`
+        // parameters.
+        //
+        // For each test vector:
+        // * Fill wallet's derived addresses with balances (as specified by test vector).
+        //    * [0..addrs_before]          => 1000sats for each address
+        //    * [addrs_before..actual_gap] => empty addresses
+        //    * [actual_gap..addrs_after]  => 1000sats for each address
+        // * Then, perform wallet sync and obtain wallet balance
+        // * Check balance is within expected range (we can compare `stop_gap` and `actual_gap` to
+        //    determine this).
+
+        // Generates wallet descriptor
+        let descriptor_of_account = |account_index: usize| -> String {
+            format!("wpkh([c258d2e4/84h/1h/0h]tpubDDYkZojQFQjht8Tm4jsS3iuEmKjTiEGjG6KnuFNKKJb5A6ZUCUZKdvLdSDWofKi4ToRCwb9poe1XdqfUnP4jaJjCB2Zwv11ZLgSbnZSNecE/{account_index}/*)")
+        };
+
+        // Amount (in satoshis) provided to a single address (which expects to have a balance)
+        const AMOUNT_PER_TX: u64 = 1000;
+
+        // [stop_gap, actual_gap, addrs_before, addrs_after]
+        //
+        // [0]     stop_gap: Passed to [`ElectrumBlockchainConfig`]
+        // [1]   actual_gap: Range size of address indexes without a balance
+        // [2] addrs_before: Range size of address indexes (before gap) which contains a balance
+        // [3]  addrs_after: Range size of address indexes (after gap) which contains a balance
+        let test_vectors: Vec<[u64; 4]> = vec![
```

Comment:
Wouldn't it be better to use `usize` here? I see that you have to cast these values in a few places

Reply:
It seems that if we do this, other variables will need casting... Should we leave it as is?

Reply:
Yeah at a quick glance I thought using usize would be fine, but if you need to cast anyway there's no reason to change it